packages/nux/src/components/guide/index.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, Children } from '@wordpress/element';
+import { Modal, KeyboardShortcuts, IconButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import PageControl from './page-control';
+import { BackButtonIcon, ForwardButtonIcon } from './icons';
+import FinishButton from './finish-button';
+
+export function Guide( { children, className, finishButtonText, isMobile, onFinish } ) {
+	const [ currentPage, setCurrentPage ] = useState( 0 );
+
+	const numberOfPages = Children.count( children );
+	const canGoBack = currentPage > 0;
+	const canGoForward = currentPage < numberOfPages - 1;
+
+	const goBack = () => {
+		if ( canGoBack ) {
+			setCurrentPage( currentPage - 1 );
+		}
+	};
+
+	const goForward = () => {
+		if ( canGoForward ) {
+			setCurrentPage( currentPage + 1 );
+		}
+	};
+
+	if ( numberOfPages === 0 ) {
+		return null;
+	}
+
+	return (
+		<Modal
+			className={ classnames( 'nux-guide', className ) }
+			shouldCloseOnClickOutside={ false }
+			onRequestClose={ onFinish }
+		>
+
+			<KeyboardShortcuts key={ currentPage } shortcuts={ {
+				left: goBack,
+				right: goForward,
+			} } />
+
+			<div className="nux-guide__container">
+
+				{ children[ currentPage ] }
+
+				{ isMobile && ! canGoForward && (
+					<FinishButton onClick={ onFinish }>
+						{ finishButtonText || __( 'Finish' ) }
+					</FinishButton>
+				) }
+
+				<div className="nux-guide__footer">
+					{ canGoBack && (
+						<IconButton
+							className="nux-guide__back-button"
+							icon={ <BackButtonIcon /> }
+							onClick={ goBack }
+						>
+							{ __( 'Previous' ) }
+						</IconButton>
+					) }
+					<PageControl
+						currentPage={ currentPage }
+						numberOfPages={ numberOfPages }
+						setCurrentPage={ setCurrentPage }
+					/>
+					{ canGoForward && (
+						<IconButton
+							className="nux-guide__forward-button"
+							icon={ <ForwardButtonIcon /> }
+							onClick={ goForward }
+						>
+							{ __( 'Next' ) }
+						</IconButton>
+					) }
+					{ ! isMobile && ! canGoForward && (
+						<FinishButton
+							className="nux-guide__finish-button"
+							onClick={ onFinish }
+						>
+							{ finishButtonText || __( 'Finish' ) }
+						</FinishButton>
+					) }
+				</div>
+
+			</div>
+
+		</Modal>
+	);
+}
+
+export function Page( props ) {
+	return <div { ...props } />;
+}
+
+const EnhancedGuide = withSelect( ( select ) => ( {
+	isMobile: select( 'core/viewport' ).isViewportMatch( '< small' ),
```

Comment:
I think we have specific HoC to do exactly that for you instead of using the store directly. `withViewpostMatch`

Reply:
Ah, perfect! Thoughts on adding `@wordpress/viewport` as a dependency to `@wordpress/components` so that we can move `Guide` into `@wordpress/components`?

Reply:
I do remember that we discussed the same thing a long time ago. I think I'm fine with it (what I don't like is the indirect dependency on the data module). That said, I feel like the whole viewport package could be just a `useViewportMatch` hook in the compose package.

Reply:
I changed `Guide` to use `useMediaQuery` and moved the component to `@wordpress/components` in 78104f3db0190d6fac295933d44bee00a5d15c9f. I like the idea of deprecating `@wordpress/viewport` in lieu of a hook in `@wordpress/compose`, but let's look into that separately.